### PR TITLE
Return original class labels without encoding

### LINKED ISSUE #369: dice-ml needs to return the original classes instead of encoded classes
We found that the tool predicts 0 and 1 in the backend by assigning a class in the target feature to a particular label. However, it is not able to show the string values, which makes it difficult for me to understand the results. 

Looks like dice-ml encodes 0 and 1 for string labels. Can we return the original labels instead of 0s and 1s.

### PR DESCRIPTION
**Pull request by amit-sharma**
Fixes #369.

Only supported for sklearn models since pytorch/tf models cannot accept non-numeric labels.

### CODE REVIEW

File: dice_ml/explainer_interfaces/dice_KD.py [hunk 1]
```diff
@@ -122,16 +121,35 @@ def _generate_counterfactuals(self, query_instance, total_CFs, desired_range=Non
...
-
+        # Decoding the output label back to original data type (e.g., str labels)
+        query_instance[self.data_interface.outcome_name] = \
+            self.decode_model_output(query_instance[self.data_interface.outcome_name])
+        if self.final_cfs_df is not None:
+            self.final_cfs_df[self.data_interface.outcome_name] = self.cfs_preds
+            self.final_cfs_df[self.data_interface.outcome_name] = \
+                self.decode_model_output(self.final_cfs_df[self.data_interface.outcome_name])
+            if self.final_cfs_df_sparse is not None:
+                self.final_cfs_df_sparse[self.data_interface.outcome_name] = \
```
Comment (gaugup) [inline dice_ml/explainer_interfaces/dice_KD.py:132]: This looks like repeated code in dice_KD.py,  dice_genetic.py and dice_random.py. Could you create a common function instead for this code?

File: tests/test_dice_interface/test_dice_genetic.py [hunk 2]
```diff
@@ -112,6 +112,37 @@ def test_predict_custom(self, desired_class, sample_custom_query_2, mocker):
...
 
+class TestDiceGeneticBinaryStrClassificationMethods:
+    @pytest.fixture(autouse=True)
+    def _initiate_exp_object(self, genetic_binary_str_classification_exp_object):
+        self.exp = genetic_binary_str_classification_exp_object  # explainer object
+
+    # When invalid desired_class is given
+    @pytest.mark.parametrize(("desired_class", "total_CFs"), [(7, 3)])
+    def test_no_cfs(self, desired_class, sample_custom_query_1, total_CFs):
+        with pytest.raises(UserConfigValidationException):
```
Comment (gaugup) [inline tests/test_dice_interface/test_dice_genetic.py:123]: Could you test the exception message encountered here?

Comment (amit-sharma) [reply tests/test_dice_interface/test_dice_genetic.py:123]: Done

File: dice_ml/explainer_interfaces/dice_KD.py (hunk 1 above)
Comment (amit-sharma) [reply dice_ml/explainer_interfaces/dice_KD.py:132]: Good idea. Done.

Comment (gaugup) [reply dice_ml/explainer_interfaces/dice_KD.py:132]: @amit-sharma have you sent an update commit on this comment?

File: tests/test_dice_interface/test_dice_genetic.py (hunk 2 above)
Comment (gaugup) [reply tests/test_dice_interface/test_dice_genetic.py:123]: @amit-sharma have you sent an update commit on this comment?

Comment (amit-sharma) [reply tests/test_dice_interface/test_dice_genetic.py]: oh yes, pushed now. Also added the multiclass test case.

File: dice_ml/explainer_interfaces/dice_KD.py (hunk 1 above)
Comment (amit-sharma) [reply dice_ml/explainer_interfaces/dice_KD.py]: pushed now.

File: tests/test_dice_interface/test_dice_genetic.py (hunk 2 above)
Comment (gaugup) [reply tests/test_dice_interface/test_dice_genetic.py]: Thank you

**Comment by gaugup** [comment]
thanks for the fix @amit-sharma. Merged.
